src/components/Patient/PatientRegistration.tsx

```diff
@@ -0,0 +1,1076 @@
+    setShowAutoFilledPincode(true);
+    setTimeout(() => {
+      setShowAutoFilledPincode(false);
+    }, 2000);
+  };
+
+  useEffect(() => {
+    const timeout = setTimeout(
+      () => handlePincodeChange(form.pincode?.toString() || ""),
+      1000,
+    );
+    return () => clearTimeout(timeout);
+  }, [form.pincode]);
+
+  const title = !patientId
+    ? t("add_details_of_patient")
+    : t("update_patient_details");
+
+  const errors = { ...feErrors, ...createPatientMutation.error };
+
+  const fieldProps = (field: keyof typeof form) => ({
+    value: form[field] as string,
+    onChange: (e: React.ChangeEvent<HTMLInputElement>) =>
+      setForm((f) => ({
+        ...f,
+        [field]: e.target.value === "" ? undefined : e.target.value,
+      })),
+  });
+
+  const selectProps = (field: keyof typeof form) => ({
+    value: (form[field] as string)?.toString(),
+    onValueChange: (value: string) =>
+      setForm((f) => ({ ...f, [field]: value })),
+  });
+
+  const handleDialogClose = (action: string) => {
+    if (action === "transfer") {
+      setShowTransferDialog(true);
+    } else if (action === "back") {
+      setShowTransferDialog(false);
+    } else {
+      setSuppressDuplicateWarning(true);
+      setShowTransferDialog(false);
+    }
+  };
+
+  const handleFormSubmit = (e: React.FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+    const errors: Record<string, string[]> = {};
+    const requiredFields: Array<keyof typeof form> = [
+      "name",
+      "phone_number",
+      "emergency_phone_number",
+      "gender",
+      "blood_group",
+      ageDob === "dob" ? "date_of_birth" : "year_of_birth",
+      "pincode",
+      "nationality",
+      "address",
+      "permanent_address",
+    ];
+
+    if (form.nationality === "India") {
+      requiredFields.push("state", "district", "local_body");
+    }
+
+    requiredFields.forEach((field) => {
+      if (!form[field]) {
+        errors[field] = errors[field] || [];
+        errors[field].push(`This field is required`);
+      } else if (
+        ageDob === "dob" &&
+        field === "date_of_birth" &&
+        !/^(19[0-9]{2}|20[0-9]{2}|2100)-(0?[1-9]|1[0-2])-(0?[1-9]|[12]\d|3[01])$/.test(
+          form[field],
+        )
+      ) {
+        errors[field] = errors[field] || [];
+        errors[field].push(t("invalid_date_format", { format: "DD-MM-YYYY" }));
+      } else if (
+        (field === "phone_number" || field === "emergency_phone_number") &&
+        form[field]?.length < 13
+      ) {
+        errors[field] = errors[field] || [];
+        errors[field].push(t("phone_number_min_error"));
+      }
+    });
+
+    if (Object.keys(errors).length > 0) {
+      setFeErrors(errors);
+      Notification.Error({
+        msg: t("please_fix_errors"),
+      });
+    } else {
+      patientId
+        ? updatePatientMutation.mutate()
+        : createPatientMutation.mutate();
+    }
+  };
+
+  const [debouncedNumber, setDebouncedNumber] = useState<string>();
```

Comment:
Do we need this, given we have the useDebounced hook?

Reply:
Atleast move all the states to the top

Reply:
The useDebounced hook implementation did not fit this use case